pkg/pillar/utils/file/file.go

```diff
@@ -38,18 +72,31 @@ func WriteRename(fileName string, b []byte) error {
 		return errors.New(errStr)
 	}
 	// Make sure the file is flused from buffers onto the disk
-	tmpfile.Sync()
+	if err := tmpfile.Sync(); err != nil {
+		errStr := fmt.Sprintf("WriteRename(%s) failed to sync temp file: %s",
+			fileName, err)
+		return errors.New(errStr)
```

Comment:
not really. If `fsync` did not happen, and power cut off happened right after close - empty file is guaranteed. Moreover, `fsync` failure would generally indicate that filesystem is in odd state (broken disk, full file system). It is likely that even periodic file system sync would fail as well, and the data would never go to the disk.